### ISSUE TITLE
Adverserial loss should call model with training argument

### ISSUE THREAD

**Issue by razorx89** [tensorflow/neural-structured-learning#28]
With regard to custom training loops, the current implementation of the adverserial loss does not call the model with a training argument. Thus when using BatchNormalization and custom training loops this leads to incorrect behaviour, since the batch statistics are not updated for the adverserial examples. Since the adverserial loss is only useful in training mode it should be safe to always call the model with `training=True`.

https://github.com/tensorflow/neural-structured-learning/blob/aad3635811d871030c1b5af4325b0f68750c5295/neural_structured_learning/keras/adversarial_regularization.py#L136

See also:
https://github.com/tensorflow/tensorflow/issues/28028
https://www.tensorflow.org/guide/migrate#customize_the_training_step

The other option is to explicitly set the learning phase in the keras backend. However, this is also discussed in the issue to be not tf-2.0 idiomatic when writing custom training loops, since it uses a hidden global state.

**Comment by csferng** [comment]
Thanks for brining up this issue and sharing the pointers.

We will add a parameter to `adversarial_loss` so that additional arguments can be passed to the model.

An interim workaround would be wrapping the model in a function, like what is done in `AdversarialRegularization`:
https://github.com/tensorflow/neural-structured-learning/blob/aad3635811d871030c1b5af4325b0f68750c5295/neural_structured_learning/keras/adversarial_regularization.py#L629
